### ISSUE TITLE
Can't get feedback.

### ISSUE THREAD

**Issue by Wrmling** [jbasen/Crestron-Hubitat-Integration#9]
Hi,

I get this massage in toolbox debugger:
Hubitat-Send_Device_Command - Error Sending Device Command: Crestron.SimplSharp.Net.Http.HttpException: HTTP/1.1 404 Not Found\x0D\x0AServer: Jetty(9.4.42.v20210604)\x0D\x0ADate: Wed, 06 Oct 2021 19:58:59 GMT\x0D\x0AContent-Type: text/html\x0D\x0AConnec(...)"    Hubitat_Communications_Err_Msg$

And don't get feedback from the hubitat.

The URL to send device events to by POST:
http%3A%2F%2F192.168.1.18%3A9000

Something wrong?

**Comment by jbasen** [comment]
Hi,

If you send me a screen-shot showing the detail view of the Hubitat Comm Manager in SimplWindows I should be able to see what is going wrong.  Please use a paint program to wipe out the Hubitat OAUTH Access Token in the screen shot as that shouldn't be shown publicly.

Thanks

**Comment by Wrmling** [comment]
![Hubitat comm](https://user-images.githubusercontent.com/83638388/136326894-1da01e9a-1803-4e11-9bd5-b3b7b45056b5.png)

Thanks for fast reply.

**Comment by Wrmling** [comment]
It's working to send commands.
I get feedback the first time i compile

**Comment by jbasen** [comment]
OK, that looks fine.  Before we get into more detailed debugging let me ask on question.  Are you using port 9000 for anything else on your Crestron processor?  The driver code sets up an http server on port 9000 (actually whatever port is specified in the Crestron _Processor_Port parameter field) to listen for update messages sent by the Hubitat.  If you aren't sure, you might try changing the Crestron _Processor_Port parameter to another arbitrary port number.  If there is a conflict, that should fix it.

A couple of other things to check:

1) If you login to your hubitat, goto apps, then go to the settings for the Maker API app.  In those settings you will see a field shown as "URL to send device events to by POST".  With the current parameters in the comm manager module that field should read:
http://192.168.1.4:9000

2) Make sure you follow exactly how the example program initializes the hubitat modules that you have placed in your simplwindows program.  It is very important that the comm manager kick off a stepper to initialize the hubitat modules related to the devices connected to your hubitat and that there be delays in that stepper.  A crestron processor can overwealm a hubitat hub if too much information is sent at too rapid a pace

3) I wrote an article for Residential Tech Today on all this.  There is a tremendous amount of information in that article that can help get all this working properly.  You can find the article here:

https://restechtoday.com/integrating-a-wide-range-of-zigbee-z-wave-and-iot-devices-with-crestron/

4) Finally, unless you are actively doing debugging you should change the Startup_Debug_Message_Output parameter on the comm manage module to "None".  You don't need the processor spending a lot of time generating debug messages and sending them to the console unless they are needed.

Let me know if any of this solves the problem you are having.  I use the modules every day in my own home so I know they work.  It is just going to take a little investigate to find out why you are having a problem.

Thanks

**Comment by Wrmling** [comment]
Perfect!
Works now when i change the "URL to send device events to by POST" manually.
But i need to do that every time i update the program.

**Comment by jbasen** [comment]
Well that is at least a step forward.  I'm trying to think why the formation of the URL to send the message to the Hubitat with the address/port of your Crestron processor isn't working properly when it works fine for me.  Are you outside of the U.S. possibly with things configured for a different language?  Again, just trying to think of what might affect how that URL is constructed in S# that might be causing a problem.

Thanks

**Comment by jbasen** [comment]
I've uplaoded v3.6.2 to GitHub.  

I haven't been able to find a solution to the problem setting the Maker API App PostURL field on a 3 series processor. So, to make this as painless as possible the code now only tries to set this field on a 4 series processor. People using running the code on a 3 series processor will have to manually set the PostURL field. The code checks if it is running on a 3 or 4 series processor and only tries to set the PostURL field on a 4 series processor. So, once a person running the code on a 3 series processor manually sets the PostURL field, they won't have to repeat that step again unless their processor IP address changes or the port they wish their processor to listen for messages changes.   

Step 15 provides information on how to manually set the PostURL field.  

I'm going to close this thread for now.  If I can figure out a work around for the setting the PostURL field on a 3 series processor I'll post a new version.  Given that my original work around for problem I encountered when originally writing this code on a 3 series processor doesn't work any more, and the cod on the 4 series processor that doesn't need the original work around won't work on a 3 series processor, I'm not holding my breath that there is anything I can do at this time to make it work. I know that isn't the best answer but it is simply the reality of the situation.
